pkg/reconciler/reconciler.go

```diff
@@ -1060,6 +1067,7 @@ func (r *Reconciler) setupWatches(mgr ctrl.Manager, c controller.Controller) err
 				obj,
 				handler.OnlyControllerOwner(),
 			),
+			secretPreds...,
```

Comment:
could we get away with fewer changes if we did something like this:

```
if err := c.Watch(
		source.Kind(
			mgr.GetCache(),
			client.Object(secret),
			handler.TypedEnqueueRequestForOwner[client.Object](
				mgr.GetScheme(),
				mgr.GetRESTMapper(),
				obj,
				handler.OnlyControllerOwner(),
			),
			preds...,
		),
	); err != nil {
		return err
	}
```

Reply:
You are right, this is better